MetasysServices.Tests/MetasysClientTests.cs

```diff
@@ -2412,12 +2414,189 @@ public void TestSendAsyncWithInvalidAbsoluteUrlThrowsException()
         {
             var requestUrl = "https://different-hostname/api/v5/networkDevices";
             var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
-            var e = Assert.Throws<MetasysHttpException>(() =>
+            var e = Assert.Throws<UriFormatException>(() =>
                 client.SendAsync(httpRequest).GetAwaiter().GetResult());
 
             PrintMessage($"TestSendAsyncWithInvalidAbsoluteUrlThrowsException: {e.Message}", true);
         }
 
+        [TestCase("v5/networkDevices", "v5/networkDevices")]
+        [TestCase("/v5/networkDevices", "v5/networkDevices")]
+        public async Task TestSendAsyncWithCustomPort(string requestUrl, string expectedRelativeUrl)
+        {
+            var originalHostname = client.Hostname;
+            var expectedUrl = $"https://{originalHostname}:8080/api/{expectedRelativeUrl}";
+
+            client.Hostname = $"{originalHostname}:8080";
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, requestUrl);
+            await client.SendAsync(httpRequest);
+
+            Assert.AreEqual(expectedUrl, httpTest.CallLog.Last().Request.RequestUri.ToString());
```

Comment:
FYI - this is technically redundant with the following assertion. I only had it because the following assertion doesn't log what the actual URL was.

Reply:
done